### DISCUSSION TITLE
Era-test-node error on MacOS (Intel CPU)

### DISCUSSION THREAD

**Discussion by Gilgamesh-818** [zkSync-Community-Hub/zksync-developers#65] [General] [answered]
When I run the cargo install command I'm getting a compile error stating that the google-cloud-default package can not be found at 

https://github.com/yoshidan/google-cloud-rust?branch=main

This is a dependency in the zksync_object_store package found in the zksync-era/core/lib repo. I'm currently trying to fix the issue locally by changing the dependency to the crate.io google-cloud-default created by yoshida. This fixes the dependency problem, but there are other errors thrown from the ClientConfig methods not being found in the gcs.rs file in the object_store's src directory. I think maybe the git google-cloud-rust fork has undergone some breaking changes and no longer works with the zksync implementation. I could be misunderstanding the issue though so some help and clarity on this would be greatly appreciated.

**Comment by idea404** [comment]
Adding repo issues for reference:

https://github.com/matter-labs/era-test-node/issues/13
https://github.com/matter-labs/era-test-node/issues/14

**Comment by MexicanAce** [comment]
Yes, we're looking into fixing the cargo install with the latest changes that were released as part of the [Boojum Release](https://zksync.mirror.xyz/HJ2Pj45EJkRdt5Pau-ZXwkV2ctPx8qFL19STM5jdYhc). Will respond to this thread (and the repo issues) when we have a resolution forward.

**Comment by dutterbutter** [comment, marked as answer]
Will be resolved in: https://github.com/matter-labs/era-test-node/pull/15

**Comment by guarinojoe** [comment]
It seems like there's an issue with the dependency on google-cloud-rust package in the zksync-era/core/lib repo. Updating the dependency to the crate.io google-cloud-default resolves the initial problem, but there are additional errors related to missing methods in the ClientConfig. This suggests potential breaking changes in the google-cloud-rust fork. Further investigation and adjustments may be needed to resolve these errors.

**Comment by tovanpanet** [comment]
It seems there might be compatibility issues between the google-cloud-rust package and the zksync-era/core/lib repo. Try checking for any updates or changes in the google-cloud-rust repository that might have caused the breakage. Alternatively, you can consider reaching out to the maintainers of both repositories for assistance or look for alternative solutions or dependencies.

**Comment by maestaspat** [comment]
It seems that the zksync-object-store package in the zkSync-era/core/lib repository is encountering dependency issues with the google-cloud-rust package. The solution involves updating the dependency to use the crate.io google-cloud-default created by yoshida. However, further investigation is needed to address the errors related to missing methods in the gcs.rs file, which may indicate breaking changes in the google-cloud-rust fork.

**Comment by sifrediblue** [comment]
The user is experiencing a compile error when running the cargo install command due to a missing dependency for the zksync_object_store package in the zkSync Era network's core/lib repository. The user has attempted to fix the issue by changing the dependency to the crate.io google-cloud-default created by yoshida, but is still encountering errors related to missing ClientConfig methods in the gcs.rs file. It is possible that the git google-cloud-rust fork has undergone breaking changes and is no longer compatible with the zkSync implementation. Further investigation and debugging may be needed to resolve these issues and determine the feasibility and impact of any necessary changes.
